homeassistant/components/shelly/update.py

```diff
@@ -0,0 +1,198 @@
+    install: Callable
+
+
+@dataclass
+class RpcUpdateDescription(
+    RpcEntityDescription, UpdateEntityDescription, RpcUpdateRequiredKeysMixin
+):
+    """Class to describe a RPC update."""
+
+
+@dataclass
+class RestUpdateDescription(
+    RestEntityDescription, UpdateEntityDescription, RestUpdateRequiredKeysMixin
+):
+    """Class to describe a REST update."""
+
+
+REST_UPDATES: Final = {
+    "fwupdate": RestUpdateDescription(
+        name="Firmware Update",
+        key="fwupdate",
+        latest_version=lambda status: status["update"]["new_version"],
+        install=lambda wrapper: wrapper.async_trigger_ota_update(),
+        device_class=UpdateDeviceClass.FIRMWARE,
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=True,
+    ),
+    "fwupdate_beta": RestUpdateDescription(
+        name="Beta Firmware Update",
+        key="fwupdate",
+        latest_version=lambda status: status["update"].get("beta_version"),
+        install=lambda wrapper: wrapper.async_trigger_ota_update(beta=True),
+        device_class=UpdateDeviceClass.FIRMWARE,
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+}
+
+RPC_UPDATES: Final = {
+    "fwupdate": RpcUpdateDescription(
+        name="Firmware Update",
+        key="sys",
+        sub_key="available_updates",
+        latest_version=lambda status: status.get("stable", {"version": None})[
+            "version"
+        ],
+        install=lambda wrapper: wrapper.async_trigger_ota_update(),
+        device_class=UpdateDeviceClass.FIRMWARE,
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=True,
+    ),
+    "fwupdate_beta": RpcUpdateDescription(
+        name="Beta Firmware Update",
+        key="sys",
+        sub_key="available_updates",
+        latest_version=lambda status: status.get("beta", {"version": None})["version"],
+        install=lambda wrapper: wrapper.async_trigger_ota_update(beta=True),
+        device_class=UpdateDeviceClass.FIRMWARE,
+        entity_category=EntityCategory.CONFIG,
+        entity_registry_enabled_default=False,
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up update entities for Shelly component."""
+    if get_device_entry_gen(config_entry) == 2:
+        return async_setup_entry_rpc(
+            hass, config_entry, async_add_entities, RPC_UPDATES, RpcUpdateEntity
+        )
+
+    if not config_entry.data[CONF_SLEEP_PERIOD]:
+        async_setup_entry_rest(
+            hass,
+            config_entry,
+            async_add_entities,
+            REST_UPDATES,
+            RestUpdateEntity,
+        )
+
+
+class RestUpdateEntity(ShellyRestAttributeEntity, UpdateEntity):
+    """Represent a REST update entity."""
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL
```

Comment:
```suggestion
    _attr_device_class = UpdateDeviceClass.FIRMWARE
    _attr_supported_features = UpdateEntityFeature.INSTALL
```

Reply:
This is already included in the EntityDescription

Reply:
Yeah I'm blind.